kie-server-parent/kie-server-api/src/test/java/org/kie/server/api/marshalling/JAXBMarshallerTest.java

```diff
@@ -0,0 +1,62 @@
+package org.kie.server.api.marshalling;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.Test;
+import org.kie.server.api.marshalling.objects.DateObject;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class JAXBMarshallerTest {
+
+    @Test
+    public void testMarshallDateObject() {
+        String expectedString = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n" +
+                "<date-object>\n" +
+                "    <localDate>2017-01-01</localDate>\n" +
+                "    <localDateTime>2017-01-01T10:10:10</localDateTime>\n" +
+                "    <localTime>10:10:10</localTime>\n" +
+                "</date-object>\n";
+
+        Set<Class<?>> extraClasses = new HashSet<Class<?>>();
+        extraClasses.add(DateObject.class);
+        Marshaller marshaller = MarshallerFactory.getMarshaller( extraClasses, MarshallingFormat.JAXB, getClass().getClassLoader() );
+
+        DateObject dateObject = new DateObject();
+        dateObject.setLocalDate( LocalDate.parse( "2017-01-01" ) );
+        dateObject.setLocalDateTime( LocalDateTime.parse("2017-01-01T10:10:10") );
+        dateObject.setLocalTime( LocalTime.parse( "10:10:10" ) );
+
+        String dateObjectString = marshaller.marshall( dateObject );
+        assertNotNull( dateObjectString );
+
+        assertEquals( expectedString, dateObjectString );
+    }
+
+    @Test
+    public void testUnmarshallDateObject() {
+        String expectedString = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n" +
+                "<date-object>\n" +
+                "    <localDate>2017-01-01</localDate>\n" +
+                "    <localDateTime>2017-01-01T10:10:10</localDateTime>\n" +
+                "    <localTime>10:10:10</localTime>\n" +
+                "</date-object>\n";
+
+        Set<Class<?>> extraClasses = new HashSet<Class<?>>();
+        extraClasses.add(DateObject.class);
+        Marshaller marshaller = MarshallerFactory.getMarshaller( extraClasses, MarshallingFormat.JAXB, getClass().getClassLoader() );
+
+        DateObject dateObject = marshaller.unmarshall( expectedString, DateObject.class );
+        assertNotNull( dateObject );
+
+        assertEquals( "2017-01-01", dateObject.getLocalDate().toString() );
```

Comment:
I might avoid the toString() (not that it's localized so it's not platform dependent iirc) althoug it's just a matter of taste:
assertEquals(LocalDate.of(2017, 1, 1), dateObject.getLocalDate());